src/main/java/com/google/sps/servlets/AddOrganizationServlet.java

```diff
@@ -52,15 +60,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     openHours.add(Integer.parseInt(hourOpen));
     openHours.add(Integer.parseInt(hourClosed));
 
-    ArrayList<String> moderatorList = new ArrayList<String>();
-    moderatorList.add("anon creator"); //dummy value for before auth gets set up
+    /* This implementation stores history entries as embedded entities instead of custom objects
+     * because it is much simpler that way */
+    ArrayList changeHistory = new ArrayList<>();
+    UserService userService = UserServiceFactory.getUserService();
+    boolean isUserLoggedIn = userService.isUserLoggedIn();
+    String username;
+    if (isUserLoggedIn) {
+      /* Currently uses user email to be consistent w other parts of codebase, subject to change */
+      username = userService.getCurrentUser().getEmail();
+    } else {
+      username = "Unknown Author";
+    }
+    
+    EmbeddedEntity historyEntry = new EmbeddedEntity();
+
+    historyEntry.setProperty("changeAuthor", username);
+    historyEntry.setProperty("changeMessage", "Organization was registered");
+    historyEntry.setProperty("changeTimeStamp", millisecondSinceEpoch);
+    changeHistory.add(historyEntry);
```

Comment:
How will this scale to other types of history additions? Is there functionality here that should be factored out?

Reply:
I figured for the functionalities we have right now it makes sense to record a message, author, and timestamp, and for more complicated tasks such as editing a moderator list we can either change the history entities to have more fields, or just append together a string message for it & follow this same format. For now, I think the strategy of just putting together a message for each change will scale up well though.

Reply:
What if someone other than you creates those pages and names the properties slightly differently? 

Looking at these fields, is there a common core that could be pulled out and work with where we're planning to go? (AKA how could you generalize this. Referring back to the design docs could be helpful here)

Reply:
In our design doc we wanted to generalize this by creating a history object and writing that into the datastore, but I found that it's much simpler to use an embedded entity than set up a customized type like that, so we decided to change the history structure to that and agree on using these property names for all history writes

Reply:
I get that it's less code in this PR to add this feature inline, but I'm not convinced it's simpler for the project as a whole. :) Agreeing on names etc is a lot more error prone than using the compiler / a type system, and it seems like you (or someone else in the pod) will immediately need to redo and generalize this once another mutating doFn is added.

Could you take a stab at a more general solution?

Reply:
I agree- I'm working on following Tay's advice below about making a HistoryManager class to take care of these things wherever it is called